mantis-control-plane/mantis-control-plane-client/src/main/java/io/mantisrx/server/master/client/MasterClientWrapper.java

```diff
@@ -238,7 +238,13 @@ public Observable<EndpointChange> getSinkLocations(final String jobId, final int
                                             for (WorkerHost host : workerAssignments.getHosts().values()) {
                                                 final int workerIndex = host.getWorkerIndex();
                                                 final int totalFromPartitions = workerAssignments.getNumWorkers();
-                                                numSinkWorkersSubject.onNext(new JobSinkNumWorkers(jobId, totalFromPartitions));
+                                                final int runningWorkers = (int) workerAssignments
+                                                    .getHosts()
+                                                    .values()
+                                                    .stream()
+                                                    .filter(e -> MantisJobState.isRunningState(e.getState()))
```

Comment:
isRunningState is true for launched, startInitiated, and started. I think we'd have the same problem as using totalWorkers during autoscaling in that case.

Reply:
good catch. I meant to use `isOnSlaveState` wdyt?